fuse_constraints/src/normal_delta_pose_2d.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020, Clearpath Robotics
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <fuse_constraints/normal_delta_pose_2d.h>
+#include <fuse_core/util.h>
+
+#include <Eigen/Core>
+#include <glog/logging.h>
+
+
+namespace fuse_constraints
+{
+
+NormalDeltaPose2D::NormalDeltaPose2D(const fuse_core::MatrixXd& A, const fuse_core::Vector3d& b) :
+  ceres::SizedCostFunction<ceres::DYNAMIC, 2, 1, 2, 1>(),
```

Comment:
@svwilliams I just tried added this here and for the `NormalPriorPose2D`, so the base class constructor is explicitly called, but the tests still fail, so it looks like the base class constructor was already called implicitly, which I believe it's what the C++ standard says.

I wonder if you have any idea what the issue could be. :thinking:

Reply:
BTW This change is in a separate commit, so we can just drop that commit if we don't want it, since it's not really doing anything different.

Reply:
If the tests pass now, do you want me to get rid of the commit that introduced this?

I don't have a strong opinion, but I'd prefer to remove it if it does nothing, because it's a bit error prone, since the template arguments must match the ones in the class definition.

Reply:
Yeah, if we don't need it, I say get rid of it.

Reply:
Done